soccer_vision_msgs/msg/Robot.msg

```diff
@@ -0,0 +1,28 @@
+# Describes a detected obstacle from vision modules, in image coordinates.
+
+uint8 NUMBER_UNKNOWN=0
+
+uint8 TEAM_UNKNOWN=0
+uint8 TEAM_OWN=1
+uint8 TEAM_OPPONENT=2
+
+# A bounding box that surrounds the obstacle in the image, this does not have to include
+# the whole obstacle, but only the part of the obstacle that was detected.
+vision_msgs/BoundingBox2D bb
+
+# The number of the robot, if it can be read from the image.
+# Values can range from NUMBER_UNKNOWN(0) up to 255.
+uint8 player_number
+
+# Which team's robot it is. 
+# Value can be TEAM_UNKNOWN, TEAM_OWN or TEAM_OPPONENT
+uint8 team
+
```

Comment:
What do you think about other information regarding the state of the detected robot. Teams may classify walking, fallen, partially visible, burning, ... robots or add an orientation to see in which direction the robot is facing.

Reply:
I like that idea! I've never used it but it could be very helpful. I'll see if I can add some fields.

Reply:
I've added a state field to the msg, but only keeping the states very vague. We could possibly add states later if needed.

Burning... :open_mouth:

Reply:
I'm not sure how to express a robot's heading in image coordinates... Robot heading seems to only make sense if you're thinking in a RobotRelative space. If a robot is not upright, how should we define heading, and perhaps the learn of the robot.

Consider a case like below. I tried describing the robot's heading (blue arrow) and robot's lean (red arrow) but doesn't seem that nice...

![Screenshot from 2021-09-25 13-47-23](https://user-images.githubusercontent.com/18237662/134758555-7e68b6c7-3a60-4796-ab11-fdf8f7f5f993.png)

Reply:
> Consider a case like below. I tried describing the robot's heading (blue arrow) and robot's lean (red arrow) but doesn't seem that nice...

Yea finding a proper definition seems quite hard for this. Maybe we could add some distinct states like `VIEW_BACK`, `VIEW_LEFT`, `VIEW_RIGHT`, `VIEW_FRONT`. This is not as general as your proposal, but it would cover the basic cases.

Reply:
I've updated both of these to be your suggestion. I've used the word "Facing" rather than "View", because I think "View" can mean many different things.